File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/types/message/digitaltwin/GetDigitalTwinListResponse.java
package org.opcfoundation.webserver.types.message.digitaltwin;

import org.opcfoundation.webserver.types.DigitalTwinDescriptor;

import java.util.ArrayList;
import java.util.List;

public class GetDigitalTwinListResponse {
    private final List<DigitalTwinDescriptor> digitalTwins;
    private boolean containsMoreData;

    public GetDigitalTwinListResponse()
    {
        this.digitalTwins = new ArrayList<>();
        this.containsMoreData = false;
    }

    public void add(DigitalTwinDescriptor descriptor)
    {
        digitalTwins.add(descriptor);
    }

    public List<DigitalTwinDescriptor> getDigitalTwins() {
        return digitalTwins;
    }

    public boolean containsMoreData()
    {
        return containsMoreData;
    }

    public void setContainsMoreData(boolean containsMoreData) {
        this.containsMoreData = containsMoreData;
    }
}

File: opcua-webservice-node/src/server/service/message/digitialtwin/GetDigitalTwinListResponse.ts
import { DigitalTwinDescriptor } from "../../../types/digitaltwin/DigitalTwinDescriptor";

export class GetDigitalTwinListResponse {
    private readonly _digitalTwins: Array<DigitalTwinDescriptor>;
    private _containsMoreData: boolean;

    constructor() {
        this._digitalTwins = [];
        this._containsMoreData = false;
    }

    add(descriptor: DigitalTwinDescriptor): void {
        this._digitalTwins.push(descriptor);
    }

    get digitalTwins(): Array<DigitalTwinDescriptor> {
        return this._digitalTwins;
    }

    get containsMoreData(): boolean {
        return this._containsMoreData;
    }

    set containsMoreData(containsMoreData: boolean) {
        this._containsMoreData = containsMoreData;
    }
}